File: Coding Challenges/DailyChallenge/Python/20260101_ResolutionStreak.py
def resolution_streak(days):
    walkGoal = 10000
    screenGoal = 120
    readGoal = 5

    streak = 0

    for dayResult in days:
        steps,screenTime,pagesRead = dayResult

        if(
            steps < walkGoal 
            or screenTime > screenGoal 
            or pagesRead < readGoal
        ):
            break

        streak += 1

    if streak == len(days): 
        return f"Resolution on track: {streak} day streak."
    else:
        return f"Resolution failed on day {streak + 1}: {streak} day streak."

File: Coding Challenges/Daily Challenge/JavaScript/20260101_ResolutionStreak.js
function resolutionStreak(days) {
  const walkGoal = 10000;
  const screenGoal = 120;
  const readGoal = 5;

  let streak = 0;

  for(let day = 0; day < days.length; day++) {
    let dayResult = days[day];

    let steps = dayResult[0];
    let screenTime = dayResult[1];
    let pagesRead = dayResult[2];

    if(steps < walkGoal || screenTime > screenGoal || pagesRead < readGoal) {
      break;
    }

    streak++;
  }

  if(streak == days.length) {
    return `Resolution on track: ${streak} day streak.`;
  }
  else {
    return `Resolution failed on day ${streak + 1}: ${streak} day streak.`;
  }
}